assets/dev/js/frontend/handlers/accessibility/nested-title-keyboard-handler.js

```diff
@@ -55,43 +72,79 @@ export default class NestedTitleKeyboardHandler extends Base {
 		return `[${ indexAttribute }="${ titleIndex }"]`;
 	}
 
+	getActiveTitleElement() {
+		const activeTitleFilter = this.getSettings( 'ariaAttributes' ).activeTitleSelector;
+		return this.elements.$itemTitles.filter( activeTitleFilter );
+	}
+
+	onInit( ...args ) {
+		super.onInit( ...args );
+	}
+
 	bindEvents() {
 		this.elements.$itemTitles.on( this.getTitleEvents() );
-		this.elements.$itemContainers.children().on( 'keydown', this.handleContentElementEscapeEvent.bind( this ) );
+		this.elements.$focusableContainerElements.on( this.getContentElementEvents() );
 	}
 
 	unbindEvents() {
 		this.elements.$itemTitles.off();
 		this.elements.$itemContainers.children().off();
 	}
 
-	handleTitleKeyboardNavigation( event ) {
-		const directionKeys = [ 'ArrowLeft', 'ArrowRight', 'ArrowUp', 'ArrowDown', 'Home', 'End' ],
-			activationKeys = [ 'Enter', 'Space' ];
+	getTitleEvents() {
+		return {
+			keydown: this.handleTitleKeyboardNavigation.bind( this ),
+		};
+	}
+
+	getContentElementEvents() {
+		return {
+			keydown: this.handleContentElementKeyboardNavigation.bind( this ),
+		};
+	}
+
+	isDirectionKey( event ) {
+		const directionKeys = [ 'ArrowLeft', 'ArrowRight', 'ArrowUp', 'ArrowDown', 'Home', 'End' ];
+		return directionKeys.includes( event.key );
+	}
 
-		if ( directionKeys.includes( event.key ) ) {
+	isActivationKey( event ) {
+		const activationKeys = [ 'Enter', ' ' ];
+		return activationKeys.includes( event.key );
+	}
+
+	handleTitleKeyboardNavigation( event ) {
+		if ( this.isDirectionKey( event ) ) {
 			event.preventDefault();
 
 			const currentTitleIndex = parseInt( this.getTitleIndex( event.currentTarget ) ) || 1,
 				numberOfTitles = this.elements.$itemTitles.length,
 				titleIndexUpdated = this.getTitleIndexFocusUpdated( event, currentTitleIndex, numberOfTitles );
 
-			this.changeTitleFocus( currentTitleIndex, titleIndexUpdated );
-		} else if ( activationKeys.includes( event.key ) ) {
+			this.changeTitleFocus( titleIndexUpdated );
+			event.stopPropagation();
+		} else if ( this.isActivationKey( event ) ) {
 			event.preventDefault();
 
+			const isLinkElement = 'a' === event.currentTarget.tagName.toLowerCase();
+
+			if ( ! elementorFrontend.isEditMode() && isLinkElement ) {
+				location.href = event.currentTarget.getAttribute( 'href' );
+				event.stopPropagation();
+			}
+
+			if ( isLinkElement ) {
+				return;
+			}
```

Comment:
Suggestion
Nested conditions.
Can this be extracted to a separate function?

Reply:
Updated